src/api/app/controllers/staging/staging_projects_controller.rb

```diff
@@ -69,4 +69,10 @@ def set_options
       @options[option] = params[option].present?
     end
   end
+
+  def set_staging_project
+    @staging_project = @project.staging.staging_projects.find_by(name: params[:staging_project_name])
```

Comment:
now also raise  *NotFound `unless @project.staging` to fix #8686 - you're so close :)

Reply:
Just to clarify: it raised a 500 error when the project did exist but it wasn't associated to a staging workflow. If the project didn't exist, it was raising a proper 404 error.

Now the first case is also raising a 404. Please have another look, @coolo.